### PR TITLE
fix: data usage crawler env handling, usage-cache.bin location

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: data usage crawler env handling, usage-cache.bin location

## Motivation and Context
canonicalize the envs such that we can bring
these envs as part of the config values,
as a subsequent change.

also fix the location of `usage-cache.bin` per bucket,
it should be inside `.minio.sys/buckets/<bucket_name>/usage-cache.bin`

this is the correct idiomatic location

## How to test this PR?
```
export MINIO_DISK_USAGE_CRAWL_ENABLE=off
``` 
To turn-off crawler

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Review by krishnasrinivas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#32a24168-6dd6-46d6-9fc6-cedfcac36bcb) |
<a name="32a24168-6dd6-46d6-9fc6-cedfcac36bcb"></a>
### 9163-9287bac/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30695
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp fbaa5624b0cc:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 11 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 16 seconds
(4/15) Running aws-sdk-ruby tests ... done in 18 seconds
(5/15) Running awscli tests ... done in 3 minutes and 31 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 18 seconds
(8/15) Running minio-dotnet tests ... done in 2 minutes and 17 seconds
(9/15) Running minio-go tests ... done in 21 minutes and 9 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 45 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11286,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-3dj3grk, objectName=minio-java-test-3moi0c5, resource=/minio-java-test-3dj3grk/minio-java-test-3moi0c5, requestId=15FD9527631E8218, hostId=2f959db7-ba11-4bb0-9f65-442299b32bda)\nrequest={method=PUT, url=http://minio-c3.minio.io:30695/minio-java-test-3dj3grk/minio-java-test-3moi0c5?uploadId=c83853045fae8f10&partNumber=1, headers=x-amz-copy-source-if-match: e6b588bc51098c806a25849db264cfa2-2\nx-amz-copy-source: minio-java-test-3dj3grk/minio-java-test-31bsh43\nAccept-Encoding: identity\nHost: minio-c3.minio.io:30695\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200319T025720Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200319/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-03-19T01-54-30Z\nVary: Origin\nX-Amz-Request-Id: 15FD9527631E8218\nX-Xss-Protection: 1; mode=block\nDate: Thu, 19 Mar 2020 02:57:20 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#7ad967af-d333-4118-96fd-9a2cc20da63f) |
<a name="7ad967af-d333-4118-96fd-9a2cc20da63f"></a>
### 9163-db7d262/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32209
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5dc736be3264:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 17 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 13 minutes and 54 seconds
(4/15) Running aws-sdk-ruby tests ... done in 35 seconds
(5/15) Running awscli tests ... done in 3 minutes and 50 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 12 minutes and 7 seconds
(8/15) Running minio-dotnet tests ... done in 4 minutes and 15 seconds
(9/15) Running minio-go tests ... done in 23 minutes and 35 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 51 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11340,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-3tlgrf0, objectName=minio-java-test-nvdu6g, resource=/minio-java-test-3tlgrf0/minio-java-test-nvdu6g, requestId=15FD9BE3B9EF4EFA, hostId=5a0730ce-68ac-4fcb-8201-3f24112b5700)\nrequest={method=PUT, url=http://minio-dev6.minio.io:32209/minio-java-test-3tlgrf0/minio-java-test-nvdu6g?uploadId=3ddad8ef4016da05&partNumber=1, headers=x-amz-copy-source-if-match: 7dd5484424cdc1fb54ce440755676008-2\nx-amz-copy-source: minio-java-test-3tlgrf0/minio-java-test-2m85h3c\nAccept-Encoding: identity\nHost: minio-dev6.minio.io:32209\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200319T050046Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200319/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 408\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-03-19T03-40-16Z\nVary: Origin\nX-Amz-Request-Id: 15FD9BE3B9EF4EFA\nX-Xss-Protection: 1; mode=block\nDate: Thu, 19 Mar 2020 05:00:46 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#d3af3d28-f135-4402-a937-1b014fc39b76) |
<a name="d3af3d28-f135-4402-a937-1b014fc39b76"></a>
### 9163-0c4278b/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:30602
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4625aef59493:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 11 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 0 seconds
(4/15) Running aws-sdk-ruby tests ... done in 19 seconds
(5/15) Running awscli tests ... done in 2 minutes and 46 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 6 minutes and 28 seconds
(8/15) Running minio-dotnet tests ... done in 2 minutes and 35 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 4 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 34 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11277,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-3h461bo, objectName=minio-java-test-1g7tevt, resource=/minio-java-test-3h461bo/minio-java-test-1g7tevt, requestId=15FD9F48A0D14777, hostId=57ed8d22-5bcc-45e4-be31-b9b7c3f39e1a)\nrequest={method=PUT, url=http://minio-c4.minio.io:30602/minio-java-test-3h461bo/minio-java-test-1g7tevt?uploadId=5201cf4f5a9ebda2&partNumber=1, headers=x-amz-copy-source-if-match: 21ba5bc6ca5831f811db3f9049b07769-2\nx-amz-copy-source: minio-java-test-3h461bo/minio-java-test-1nd7ac8\nAccept-Encoding: identity\nHost: minio-c4.minio.io:30602\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200319T060258Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200319/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-03-19T05-01-50Z\nVary: Origin\nX-Amz-Request-Id: 15FD9F48A0D14777\nX-Xss-Protection: 1; mode=block\nDate: Thu, 19 Mar 2020 06:02:58 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#116ffc60-81ed-4a65-bd76-eac883910c60) |
<a name="116ffc60-81ed-4a65-bd76-eac883910c60"></a>
### 9163-3ce463a/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:30898
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d0691d03c34f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 6 minutes and 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 2 minutes and 46 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 18 seconds
(8/15) Running minio-dotnet tests ... done in 2 minutes and 56 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 30 seconds
(10/15) Running minio-java tests ... FAILED in 7 minutes and 48 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11159,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-2s32dld, objectName=minio-java-test-2t7ma02, resource=/minio-java-test-2s32dld/minio-java-test-2t7ma02, requestId=15FDA29A99B3D284, hostId=b4602a8c-08a4-4189-a459-e37d89b456ae)\nrequest={method=PUT, url=http://minio-dev3.minio.io:30898/minio-java-test-2s32dld/minio-java-test-2t7ma02?uploadId=0cec1782e1e78aa6&partNumber=1, headers=x-amz-copy-source-if-match: 8e2001b78880bb3b1efe8f6bc3f956c0-2\nx-amz-copy-source: minio-java-test-2s32dld/minio-java-test-1nue6u6\nAccept-Encoding: identity\nHost: minio-dev3.minio.io:30898\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200319T070348Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200319/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-03-19T06-03-19Z\nVary: Origin\nX-Amz-Request-Id: 15FDA29A99B3D284\nX-Xss-Protection: 1; mode=block\nDate: Thu, 19 Mar 2020 07:03:49 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#c4d5e856-eec0-4c29-b554-41728abba96e) |
<a name="c4d5e856-eec0-4c29-b554-41728abba96e"></a>
### 9163-9df2f55/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:32460
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 46d148ebb63d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 6 minutes and 47 seconds
(4/15) Running aws-sdk-ruby tests ... done in 17 seconds
(5/15) Running awscli tests ... FAILED in 54 seconds
{
  "name": "awscli",
  "duration": 1480,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}

Executed 4 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: docs/config/README.md [hunk 1]
```diff
@@ -235,8 +235,17 @@ This behavior is consistent across all keys, each key self documents itself with
...
-Enable this to turn on Write-Once-Read-Many. By default it is set to `off`. Set ``MINIO_WORM=on`` environment variable to enable WORM mode.
+#### Usage crawler
+Data usage crawler is enabled by default, following ENVs allow for more staggered delay in terms of usage calculation.
+
+Example: Add a delay factor for crawler to sleep between crawling. By default this value is 10 microseconds, allowed maximum is 1 second. Following example sets this value to 30 microseconds.
+```sh
+export MINIO_DISK_USAGE_CRAWL_DELAY=30
+minio server /data
+```
+
```
Comment (klauspost) [inline docs/config/README.md]:
```suggestion
Data usage crawler is enabled by default, following ENVs allow for more staggered delay in terms of usage calculation. The crawler can be completely disabled by setting `MINIO_DISK_USAGE_CRAWL_ENABLE=off`. This of course means that data usage will no longer be updated.

The crawler adapts to the system speed and completely pauses when the system is under load. It is possible to adjust the speed of the crawler and thereby the latency of updates being reflected. 

The delays between each operation of the crawl can be adjusted by the `MINIO_DISK_USAGE_CRAWL_DELAY` environment variable. By default the value is `10`. This means the crawler will sleep 10x the time each operation takes. This will in most setups make the crawler slow enough to not impact overall system performance.

Setting `MINIO_DISK_USAGE_CRAWL_DELAY` to a *lower* value will make the crawler faster and setting it to 0 will make the crawler run at full speed (not recommended). Setting it to a higher value will make the crawler slower, but consume less resources.

Example:
'''sh
export MINIO_DISK_USAGE_CRAWL_DELAY=30
minio server /data
'''

This will decrease the crawler speed by a factor of 3, reducing the system resource use, but increasing the latency of updates being reflected in the totals.

```

Comment (klauspost) [reply docs/config/README.md]: NOTE, the suggesting thingie seems broken. I had to replace the example backticks for it to work :/

File: cmd/data-usage-cache.go [hunk 2]
```diff
@@ -114,7 +113,7 @@ func (d *dataUsageCache) find(path string) *dataUsageEntry {
 func (d *dataUsageCache) dui(path string, buckets []BucketInfo) DataUsageInfo {
 	e := d.find(path)
 	if e == nil {
-		return DataUsageInfo{LastUpdate: time.Now()}
+		return DataUsageInfo{LastUpdate: UTCNow()}
```
Comment (klauspost) [inline cmd/data-usage-cache.go:116]: Not technically needed. The issue is with the test, but if this fixes it, no problem.

(just resolve)

File: cmd/data-usage.go [hunk 3]
```diff
@@ -51,11 +50,9 @@ const (
 
 // initDataUsageStats will start the crawler unless disabled.
 func initDataUsageStats() {
-	dataUsageEnabled, err := config.ParseBool(env.Get(dataUsageCrawlConf, config.EnableOn))
-	if err == nil && !dataUsageEnabled {
-		return
+	if env.Get(envDataUsageCrawlConf, config.EnableOn) == config.EnableOn {
```
Comment (klauspost) [inline cmd/data-usage.go:53]: Technically this change breaks compatibility with the existing behavior.

**Review by klauspost — CHANGES_REQUESTED** [review]

**Comment by klauspost** [comment]
This does AFAICT not change `usage-cache.bin location` as claimed in the title.

File: cmd/data-usage.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/data-usage.go:53]: Yes it's all intentional, no one is using it yet.

**Comment by harshavardhana** [comment]
The usage-cache.bin location change is for each buckets @klauspost in xl-v1.go

File: docs/config/README.md (hunk 1 above)
Comment (harshavardhana) [reply docs/config/README.md]: Personally do not want to encourage disabling crawler. We may even drop that in future, so we should avoid explicitly documenting them. It was intentionallly left out.

File: cmd/xl-v1.go [hunk 4]
```diff
@@ -321,11 +319,9 @@ func (xl xlObjects) crawlAndGetDataUsage(ctx context.Context, buckets []BucketIn
 					return
 				default:
 				}
-				if dataUsageDebug {
-					logger.Info(color.Green("crawlAndGetDataUsage:")+" Scanning bucket %v.", bucket.Name)
-				}
+
 				// Load cache for bucket
-				cacheName := path.Join(dataUsageBucketCacheDir, bucket.Name+".bin")
+				cacheName := pathJoin(bucket.Name, dataUsageCacheName)
```
Comment (klauspost) [inline cmd/xl-v1.go:324]: Why this change? A bucket called `usage-cache` would conflict with the global usage cache. And having it in a separate dir will make it much easier to manage.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#c9e5d748-863f-4d2e-8968-65c20a549ec0) |
<a name="c9e5d748-863f-4d2e-8968-65c20a549ec0"></a>
### 9163-df12e29/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31816
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 65e07319fe2a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 8 seconds
(4/15) Running aws-sdk-ruby tests ... done in 19 seconds
(5/15) Running awscli tests ... done in 3 minutes and 22 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 17 seconds
(8/15) Running minio-dotnet tests ... done in 2 minutes and 21 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 39 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 38 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11351,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-3bpm4nk, objectName=minio-java-test-31tfelm, resource=/minio-java-test-3bpm4nk/minio-java-test-31tfelm, requestId=15FDB12628D1F36D, hostId=a61dd704-882c-4565-8719-679de5e0701f)\nrequest={method=PUT, url=http://minio-c3.minio.io:31816/minio-java-test-3bpm4nk/minio-java-test-31tfelm?uploadId=bfc60701c524bed1&partNumber=1, headers=x-amz-copy-source-if-match: c1e632b59f756ac6956ff0c870c52ec6-2\nx-amz-copy-source: minio-java-test-3bpm4nk/minio-java-test-k9t5bm\nAccept-Encoding: identity\nHost: minio-c3.minio.io:31816\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200319T113021Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200319/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-03-19T10-28-26Z\nVary: Origin\nX-Amz-Request-Id: 15FDB12628D1F36D\nX-Xss-Protection: 1; mode=block\nDate: Thu, 19 Mar 2020 11:30:21 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]
ok, let's go for it 👍

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#8fe3f148-6505-4ef1-8bee-1157b4351dc4) |
| mint-gateway-azure.sh | :x: [more...](#04994a22-2b1f-44ca-ab0e-0559fb0652f1) |
<a name="04994a22-2b1f-44ca-ab0e-0559fb0652f1"></a>
### 9163-927a94e/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:31300
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9150d5bc3522:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 9 seconds
(4/15) Running aws-sdk-ruby tests ... done in 18 seconds
(5/15) Running awscli tests ... FAILED in 1 minutes and 49 seconds
{
  "name": "awscli",
  "duration": 1423,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}

Executed 4 out of 15 tests successfully.
```
<a name="8fe3f148-6505-4ef1-8bee-1157b4351dc4"></a>
### 9163-927a94e/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31610
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 03e4ac6aaeb9:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 18598,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "remove_bucket failed: s3://awscli-mint-test-bucket-25687 An error occurred (XMinioServerNotInitialized) when calling the DeleteBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 4 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
